vertical-pod-autoscaler/updater/apimock/vpa_api_mock.go

```diff
@@ -0,0 +1,182 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package apimock contains temporary definitions of Vertical Pod Autoscaler related objects - to be replaced with real implementation
+// Definitions based on VPA design doc : https://github.com/kubernetes/community/pull/338
+package apimock
+
+import (
+	"math/rand"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	apiv1 "k8s.io/kubernetes/pkg/api/v1"
+)
+
+// VerticalPodAutoscaler Represents Vertical Pod Autoscaler configuration - to be replaced by real implementation
+type VerticalPodAutoscaler struct {
+	// Specification
+	Spec Spec
+	// Current state of VPA
+	Status Status
+}
+
+// Spec holds Vertical Pod Autoscaler configuration
+type Spec struct {
+	// Defines which pods this Autoscaler should watch and update
+	Target Target
+	// Policy for pod updates
+	UpdatePolicy UpdatePolicy
+	// Policy for container resources updates
+	ResourcesPolicy ResourcesPolicy
+}
+
+// Status holds current Vertical Pod Autoscaler state
+type Status struct {
+	// Recommended resources allocation
+	Recommendation Recommendation
+}
+
+// Target Specifies pods to be managed by Vertical Pod Autoscaler
+type Target struct {
+	// Pod labels to match
+	MatchLabels map[string]string
+}
+
+// UpdatePolicy defines policy for Pod updates
+type UpdatePolicy struct {
+	// Mode for update policy
+	Mode Mode
+}
+
+// Mode of update policy
+type Mode struct {
+}
+
+// ResourcesPolicy represents Resources allocation policy
+type ResourcesPolicy struct {
+	// Resources allocation policy for containers
+	Containers []ContainerPolicy
+}
+
+// Recommendation represents resources allocation recommended by Vertical Pod Autoscaler
+type Recommendation struct {
+	// Recommended resources allocation for containers
+	Containers []ContainerRecommendation
+}
+
+// ContainerPolicy hold resources allocation policy for single container
+type ContainerPolicy struct {
+	// Name of the container
+	Name string
+	// Memory allocation policy
```

Comment:
Not sure If you remember but we agreed that we would rather use maps for passing resources to be able to support a new resource easier. Please replace it with map - ResourceType -> Policy.